### PR TITLE
Repo org refactor

### PR DESCRIPTION
**Pull request by runenilsenoe**

### CODE REVIEW

**Comment by runenilsenoe** [comment]
Concept:

- Separating the build files from source code. recommended in many repo structure concepts
- Separating Silo hosting and using orleans terminology (This will pretty much remain untouched) the only updates to the silo project in the future(except hosting config) is additions to DI services, so if they are separated into an extension as well, the hosting project will remain clean and easy to read.
- Specify Grains and Graininterfaces by name so that they can be differentiated from regular classes easier. there is some practices that differentiate with grains. tasks should always be quick as an example. So knowing that you are in a grain explicitly might help. 
- Grain State models in grains project and shared models in interface project. (shared models can be moved to a separate project in the future if need be. Side Note: if you want to ensure smooth backwards compatibility on all levels, the grain state object parameters should be nullable. That is not a prerequisite in orleans 7 due to state versioning, but its nice to have to avoid state corruption (That way the serialisation wont crash if the object is null)

I added Orleans Dash because its freakin nice. and fun to watch. and useful to avoid bottlenecks.

**Comment by akselkvitberg** [comment]
I don't think we gain anything except complexity by running a separate Orleans Silo alongside a ASP.NET Core API.
Having the Orleans silo run locally also ensures that more grain calls will happen locally instead of over the wire, which helps with performance. 
And we would only need to deploy one container instead of two.

File: src/Grains/GrainStateModels/FailedMessageData.cs
Comment (akselkvitberg) [inline src/Grains/GrainStateModels/FailedMessageData.cs:1]: I'm not sure I like having all grain state classes in a separate folder. They should only be used by one grain, and should preferably be kept in the same file as the grain, possibly as a nested class even?

File: src/Grains/Grains.csproj [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+    <PropertyGroup>
+        <TargetFramework>net7.0</TargetFramework>
+        <ImplicitUsings>enable</ImplicitUsings>
+        <Nullable>enable</Nullable>
+    </PropertyGroup>
+
+    <ItemGroup>
+      <PackageReference Include="Microsoft.Orleans.Core.Abstractions" Version="7.1.2" />
```
Comment (akselkvitberg) [inline src/Grains/Grains.csproj:10]: Use Microsoft.Orleans.Sdk, as per Orleans documentation

File: src/Grains/GrainStateModels/FailedMessageData.cs
Comment (runenilsenoe) [reply src/Grains/GrainStateModels/FailedMessageData.cs:1]: I prefer having the grain state object itself in a separate folder in the grains lib, and then all the datamodels for its variables in grain interfaces. If this is indeed to be as lightweight as possible, a good practice is to share variable classes with the api. 

having multiple classes in the same file gets messy really quick. as soon as you go from PoC to real life grains, they tend to get huge.

File: src/Grains/Grains.csproj (hunk 1 above)
Comment (runenilsenoe) [reply src/Grains/Grains.csproj:10]: ahh, thats neat. havent used version 7 much yet. i like it

**Comment by runenilsenoe** [comment]
> I don't think we gain anything except complexity by running a separate Orleans Silo alongside a ASP.NET Core API. Having the Orleans silo run locally also ensures that more grain calls will happen locally instead of over the wire, which helps with performance. And we would only need to deploy one container instead of two.

Dont think there is much performance difference but i see your point. one container is nice. If the project is scaled in the future, one could argue it simplifies deployment to avoid the orleans deployment everytime you do changes to the api. Silos is a bit heavier to deploy. Would probably be nice as a PoC anyways. ill commit some changes

**Comment by runenilsenoe** [comment]
@akselkvitberg i see an interface is persisted in state for failed messages. i believe the correct method of storing references to grains is by id, then referencing by id with getgrain. at least that is whats recommended by the orleans-bois

**Comment by akselkvitberg** [comment]
What's the difference? It is serialized as a grain id anyways, right?

**Comment by runenilsenoe** [comment]
> What's the difference? It is serialized as a grain id anyways, right?

did not think so.

**Comment by runenilsenoe** [comment]
> What's the difference? It is serialized as a grain id anyways, right?

https://github.com/dotnet/orleans/issues/6835#issuecomment-846322787

**Comment by akselkvitberg** [comment]
Seems reasonable then

**Pull request merged**